packages/builder/src/builderStore/index.js

```diff
@@ -69,7 +69,14 @@ export const selectedComponent = derived(
     if (!$selectedScreen || !$store.selectedComponentId) {
       return null
     }
-    return findComponent($selectedScreen?.props, $store.selectedComponentId)
+    const selected = findComponent(
+      $selectedScreen?.props,
+      $store.selectedComponentId
+    )
+
+    const clone = selected ? Object.assign({}, selected) : selected
```

Comment:
`Object.assign` still creates a shallow copy so this wouldn't be safe if we're mutating any nested properties as part of `migrateSettings` (which I don't think we are for now, but obviously could in future). Maybe safer to do a proper deep clone here to ensure we'll always be safe?